EIPS/eip-4844.md

```diff
@@ -331,41 +266,33 @@ The actual `data_fee` as calculated via `calc_data_fee` is deducted from the sen
 
 ### Networking
 
-Nodes must not automatically broadcast blob transactions to their peers.
-Instead, those transactions are only announced using `NewPooledTransactionHashes` messages, and can then be manually requested via `GetPooledTransactions`.
-
-Transactions are presented as `TransactionType || TransactionNetworkPayload` on the execution layer network,
-the payload is a SSZ encoded container:
+Blob transactions have two network representations. During transaction gossip responses (`PooledTransactions`), the EIP-2718 `TransactionPayload` of the blob transaction is wrapped to become:
 
-```python
-class BlobTransactionNetworkWrapper(Container):
-    tx: SignedBlobTransaction
-    # KZGCommitment = Bytes48
-    blob_kzgs: List[KZGCommitment, MAX_TX_WRAP_KZG_COMMITMENTS]
-    # BLSFieldElement = uint256
-    blobs: List[Vector[BLSFieldElement, FIELD_ELEMENTS_PER_BLOB], LIMIT_BLOBS_PER_TX]
-    # KZGProof = Bytes48
-    kzg_aggregated_proof: KZGProof
+```
+rlp([blob_tx_payload, blob_kzgs, blobs, kzg_aggregated_proof])
 ```
 
-We do network-level validation of `BlobTransactionNetworkWrapper` objects as follows:
+Each of these elements are defined as follows:
 
-```python
-def validate_blob_transaction_wrapper(wrapper: BlobTransactionNetworkWrapper):
-    versioned_hashes = wrapper.tx.message.blob_versioned_hashes
-    commitments = wrapper.blob_kzgs
-    blobs = wrapper.blobs
-    # note: assert blobs are not malformatted
-    assert len(versioned_hashes) == len(commitments) == len(blobs)
-    assert len(versioned_hashes) > 0
-
-    # Verify that commitments match the blobs by checking the KZG proof
-    assert verify_aggregate_kzg_proof(blobs, commitments, wrapper.kzg_aggregated_proof)
-
-    # Now that all commitments have been verified, check that versioned_hashes matches the commitments
-    for versioned_hash, commitment in zip(versioned_hashes, commitments):
-        assert versioned_hash == kzg_to_versioned_hash(commitment)
-```
+- `blob_tx_payload` - standard EIP-2718 blob transaction `TransactionPayload`
+- `blob_kzgs` - list of `KZGCommitment`
+- `blobs` - list of `blob` where `blob` is a list of `BLSFieldElement`
+- `kzg_aggregated_proof` - `KZGProof`
+
+The node MUST validate `blob_tx_payload` and verify the wrapped data against it. To do so, ensure that:
+
+- `blob_tx_payload.blob_versioned_hashes` must not be empty
+- All hashes in `blob_tx_payload.blob_versioned_hashes` must start with the byte `BLOB_COMMITMENT_VERSION_KZG`
+- There must be at most `MAX_DATA_GAS_PER_BLOCK // DATA_GAS_PER_BLOB` total blob commitments in the transaction.
+- There are an equal number of versioned hashes, kzg commitments, and blobs.
+- The KZG commitments hash to the versioned hashes, i.e. `kzg_to_versioned_hash(kzg[i]) == versioned_hash[i]`
+- The KZG commitments match the blob contents. (Note: this can be optimized with additional data, using a proof for a
+  random evaluation at two points derived from the commitment and blob data)
```

Comment:
```suggestion
- The KZG commitments match the blob contents. (Note: this can be optimized using `blob_kzg_proofs`, using a proof for a
  random evaluation at a point derived from the commitment and blob data for each blob)
```